src/lib/moderatePost_wrapped.ts

```diff
@@ -2,18 +2,72 @@ import {
   AppBskyEmbedRecord,
   AppBskyEmbedRecordWithMedia,
   moderatePost,
+  AppBskyActorDefs,
+  AppBskyFeedPost,
+  AppBskyRichtextFacet,
+  AppBskyEmbedImages,
 } from '@atproto/api'
 
 type ModeratePost = typeof moderatePost
 type Options = Parameters<ModeratePost>[1] & {
   hiddenPosts?: string[]
+  mutedWords?: AppBskyActorDefs.MutedWord[]
+}
+
+export function hasMutedWord(
+  mutedWords: AppBskyActorDefs.MutedWord[],
+  text: string,
+  facets?: AppBskyRichtextFacet.Main[],
+  outlineTags?: string[],
+) {
+  const tags = ([] as string[])
+    .concat(outlineTags || [])
+    .concat(
+      facets
+        ?.filter(facet => {
+          return facet.features.find(feature =>
+            AppBskyRichtextFacet.isTag(feature),
+          )
+        })
+        .map(t => t.features[0].tag as string) || [],
+    )
+    .map(t => t.toLowerCase())
+
+  const words = text.toLowerCase().split(/\s+/)
+
+  for (const mute of mutedWords) {
+    const mutedWord = mute.value.toLowerCase()
+
+    // `content` applies to tags as well
+    if (tags.includes(mutedWord)) return true
+    // rest of the checks are for `content` only
+    if (!mute.targets.includes('content')) continue
+    // single character, has to use includes
+    if (mutedWord.length === 1 && text.includes(mutedWord)) return true
+    // too long
+    if (mutedWord.length > text.length) continue
+    // exact match
+    if (mutedWord === text) return true
+
+    // check individual character groups
+    for (const word of words) {
+      if (word === mutedWord) return true
+
+      const wordNoPunc = word.replace(/\p{P}+$/gu, '')
+      const mutedWordNoPunc = mutedWord.replace(/\p{P}+$/gu, '')
+
+      if (mutedWordNoPunc === wordNoPunc) return true
+    }
+  }
```

Comment:
If we want to enable support for phrases, I think what we can do is this:

- Instead of splitting the text into words, run your punctuation strip on it
- Convert the muteword into a regex like this (havent tested this regex): `new RegExp('[\s\t\n\r\f\,\.\;\!\?\_\-\"\(\)]' + stripPunctuation(muteWord) + '[\s\t\n\r\f\,\.\;\!\?\_\-\"\(\)]')`
- Just run the muteword regexes against the punctuation-stripped post text

Reply:
Here's how Mastodon is more or less doing its matching:

https://github.com/mary-ext/langit/blob/70cf8ed820d27337370039f517f4968c6553cd11/app/desktop/components/settings/settings-views/keyword-filters/KeywordFilterFormView.tsx#L258

Reply:
Thanks for the link! Will have a look and compare 🙏